packages/oauth/src/form/form-service.ts

```diff
@@ -201,6 +233,16 @@ export class FormService implements OAuthProtoService {
         else log.error(err)
       }
 
+      //
+      // If token failed to be decoded then try and see if
+      // it is an openshift cluster and ask the API for the username
+      //
+      if (subject === this.userProfile.getToken()) {
```

Comment:
Also we'd like to have a guard whether it's on OpenShift or not before sending an unnecessary request to the API server.